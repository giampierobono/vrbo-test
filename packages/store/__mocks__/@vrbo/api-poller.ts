// tslint:disable-next-line:no-var-requires
const apiPollerStore = require("../../../../__mocks__/@vrbo/api-poller");

module.exports = apiPollerStore;
