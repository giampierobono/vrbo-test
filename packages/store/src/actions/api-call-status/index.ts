export * from "./api-call-status.actions";
