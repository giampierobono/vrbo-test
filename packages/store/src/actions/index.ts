export * from "./api-call-status";
export * from "./poller-config";
export * from "./properties-sort";
export * from "./property";
