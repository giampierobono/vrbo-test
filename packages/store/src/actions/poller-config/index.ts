export * from "./poller-config.actions";
