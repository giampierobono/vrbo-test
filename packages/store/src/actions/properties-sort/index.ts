export * from "./properties-sort.actions";
