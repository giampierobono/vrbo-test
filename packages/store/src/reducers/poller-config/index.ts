export * from "./poller-config.reducer";
