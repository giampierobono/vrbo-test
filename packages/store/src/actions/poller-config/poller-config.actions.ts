import { ActionWithPayload, PollerConfig } from "@vrbo/data-models";

export enum PollerConfigActions {
  SetNewPollerConfig = "[PollerConfig] Set new poller config"
}

export const setNewPollerConfigAction = (
  newConfig: PollerConfig
): ActionWithPayload<PollerConfig> => ({
  type: PollerConfigActions.SetNewPollerConfig,
  payload: newConfig
});
