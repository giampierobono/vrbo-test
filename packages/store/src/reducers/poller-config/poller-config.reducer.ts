import {
  ActionWithPayload,
  PollerConfig,
  PropertyType
} from "@vrbo/data-models";
import { PollerConfigActions } from "../../actions";

export interface PollerConfigState {
  config: PollerConfig;
}

export const initialPollerConfigState: PollerConfigState = {
  config: {
    limit: 10,
    type: PropertyType.Houses
  }
};

export const pollerConfigReducer = (
  state: PollerConfigState = initialPollerConfigState,
  action: ActionWithPayload<PollerConfig>
) => {
  switch (action.type) {
    case PollerConfigActions.SetNewPollerConfig: {
      return {
        ...state,
        config: {
          ...action.payload
        }
      };
    }
    default: {
      return state;
    }
  }
};
