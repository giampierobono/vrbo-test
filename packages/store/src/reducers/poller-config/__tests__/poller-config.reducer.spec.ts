import { PropertyType } from "@vrbo/data-models";
import { setNewPollerConfigAction } from "../../../actions/poller-config";
import {
  initialPollerConfigState,
  pollerConfigReducer
} from "../poller-config.reducer";

describe("Poller config reducer", () => {
  it("should return state for UNKNOWN actions", () => {
    const result = pollerConfigReducer(initialPollerConfigState, {
      type: "UNKNOWN"
    });
    expect(result).toEqual(initialPollerConfigState);
  });

  it("should set new poller config", () => {
    const result = pollerConfigReducer(
      initialPollerConfigState,
      setNewPollerConfigAction({ limit: 100, type: PropertyType.Condos })
    );
    expect(result).toMatchSnapshot();
  });
});
