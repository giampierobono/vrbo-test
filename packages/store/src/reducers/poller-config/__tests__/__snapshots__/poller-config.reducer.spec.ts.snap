// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poller config reducer should set new poller config 1`] = `
Object {
  "config": Object {
    "limit": 100,
    "type": "condos",
  },
}
`;
